snapshot/dump.go

```diff
@@ -0,0 +1,199 @@
+package snapshot
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/flanksource/incident-commander/api"
+	"github.com/flanksource/incident-commander/db"
+	"github.com/flanksource/incident-commander/utils"
+)
+
+func getColumnNames(table string) (string, error) {
+	var columns string
+	err := db.Gorm.Raw(`SELECT string_agg(column_name, ',') from information_schema.columns where table_name = ?`, table).
+		Scan(&columns).Error
+
+	if err != nil {
+		return "", err
+	}
+	return columns, nil
+}
+
+func generateCSVDumpQuery(table, columns, idField, whereClause string) string {
+	return fmt.Sprintf(`
+        SELECT %s FROM %s WHERE %s IN (%s)
+    `, columns, table, idField, whereClause)
+}
+
+func dumpTable(table string, idField, whereClause string, ids []string, csvDirectory string) error {
+	var rows []map[string]any
+	columnNames, err := getColumnNames(table)
+	if err != nil {
+		return err
+	}
+	query := generateCSVDumpQuery(table, columnNames, idField, whereClause)
+	err = db.Gorm.Raw(query, ids).Scan(&rows).Error
+	if err != nil {
+		return err
+	}
+
+	return writeToCSVFile(csvDirectory, table+".csv", columnNames, rows)
+}
+
+func dumpComponents(directory string, componentIDs []string) error {
+	if len(componentIDs) == 0 {
+		return nil
+	}
+	var jsonBlobs []map[string]any
+
+	endpoint, err := url.JoinPath(api.CanaryCheckerPath, "/api/topology")
```

Comment:
Rather use the sdk for this: https://github.com/flanksource/incident-commander/blob/main/rules/job.go#L19